File: client-api/src/main/java/com/flipkart/grayskull/models/SecretValue.java
package com.flipkart.grayskull.models;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * Represents a secret value retrieved from Grayskull.
 * <p>
 * Contains the actual secret data including both public and private parts,
 * along with version information.
 * </p>
 */
@Getter
@AllArgsConstructor(onConstructor = @__(@JsonCreator))
@JsonIgnoreProperties(ignoreUnknown = true)
public final class SecretValue {

    /**
     * Data version number.
     */
    private final int dataVersion;

    /**
     * Public part of the secret.
     */
    private final String publicPart;

    /**
     * Private/sensitive part of the secret.
     */
    private final String privatePart;
}

File: clients/go/client-api/models/secretValue.go
package models

// SecretValue represents a secret value with versioning
type SecretValue struct {
	// DataVersion represents the version number of the secret data
	DataVersion int `json:"dataVersion"`

	// PublicPart contains the public part of the secret
	PublicPart string `json:"publicPart"`

	// PrivatePart contains the private/sensitive part of the secret
	PrivatePart string `json:"privatePart,omitempty"`
}
